front/app/modules/commercial/insights/admin/containers/Insights/Edit/InputsTable/InputsTable.test.tsx

```diff
@@ -366,15 +374,15 @@ describe('Insights Input Table', () => {
             [mockCategoriesData[1].id]
           );
         });
-        it('has an unassign button that works as expected', () => {
+        it('has an unassign button that works as expected', async () => {
           expect(
             screen
               .getAllByTestId('insightsInputsTableRow')
               .map((row) => within(row).getByRole('checkbox'))
               .map((box: any) => box.checked)
           ).toEqual([true, false]);
 
-          act(() => {
+          await act(async () => {
```

Comment:
Slightly puzzled with this, maybe I just don't understand this rtl specific part yet, why did these tests pass before you moved them to async ?

Reply:
I added the async here because I was seeing warnings about it in the test console. I think it's not necessarily that the tests won't pass but that the results might be unreliable if you don't use the async `act` when the state update is async - thus the warning.
As for `act` itself, it's a React utility that rtl re-exposes. Here is one article that really helped me understand it and the difference between the sync and the async one, if you are also curious about it: https://medium.com/@bmb21/reacts-sync-and-async-act-caa297b658b0